float64 trust
float64[] pr_data
float64[] ph_data
float64 pr
float64 ph